File: tests/human/x/php/group_by_join.php
<?php
$customers = [
    ["id" => 1, "name" => "Alice"],
    ["id" => 2, "name" => "Bob"],
];
$orders = [
    ["id" => 100, "customerId" => 1],
    ["id" => 101, "customerId" => 1],
    ["id" => 102, "customerId" => 2],
];
$groups = [];
foreach ($orders as $o) {
    foreach ($customers as $c) {
        if ($o['customerId'] === $c['id']) {
            $groups[$c['name']][] = $o;
        }
    }
}
ksort($groups);
$stats = [];
foreach ($groups as $name => $ordersFor) {
    $stats[] = ['name'=>$name, 'count'=>count($ordersFor)];
}
var_dump("--- Orders per customer ---");
foreach ($stats as $s) {
    var_dump($s['name'], "orders:", $s['count']);
}
?>

File: tests/human/x/dart/group_by_join.dart
void main() {
  var customers = [
    {'id': 1, 'name': 'Alice'},
    {'id': 2, 'name': 'Bob'},
  ];
  var orders = [
    {'id': 100, 'customerId': 1},
    {'id': 101, 'customerId': 1},
    {'id': 102, 'customerId': 2},
  ];

  var joined = [];
  for (var o in orders) {
    var customer = customers.firstWhere((c) => c['id'] == o['customerId']);
    joined.add({'name': customer['name'], 'orderId': o['id']});
  }

  var groups = <String, List<Map<String, dynamic>>>{};
  for (var j in joined) {
    groups.putIfAbsent(j['name'], () => []).add(j);
  }

  var stats = [];
  groups.forEach((name, list) {
    stats.add({'name': name, 'count': list.length});
  });

  print('--- Orders per customer ---');
  for (var s in stats) {
    print('${s['name']} orders: ${s['count']}');
  }
}
